source/ada/lsp-ada_documents.adb

```diff
@@ -718,19 +1035,39 @@ package body LSP.Ada_Documents is
          --  diff for a part of the document
 
          Out_Span := Self.To_LSP_Range (Out_Sloc);
-         Diff
-           (Self,
-            VSS.Strings.Conversions.To_Virtual_String (S.all),
-            Span,
-            Out_Span,
-            Edit);
+
+         --  Use line diff if the range is too wide
+         if Span.last.line - Span.first.line > 5 then
+            Diff
+              (Self,
+               VSS.Strings.Conversions.To_Virtual_String (S.all),
+               Span,
+               Out_Span,
+               Edit);
+         else
+            declare
+               Formatted : constant VSS.Strings.Virtual_String :=
+                 VSS.Strings.Conversions.To_Virtual_String (S.all);
+               Slice     : VSS.Strings.Virtual_String;
+
+            begin
+               LSP.Lal_Utils.Span_To_Slice (Formatted, Out_Span, Slice);
+
+               Diff_Symbols
```

Comment:
Could we also use it in the Range_Formatting action (where it makes even more sense)?